src/main/java/spoon/support/reflect/declaration/CtElementImpl.java

```diff
@@ -536,4 +541,12 @@ public CtElement clone() {
 		rh.setValue(this, value);
 		return (E) this;
 	}
+
+	public CtPath getPath() {
+		try {
+			return new CtElementPathBuilder().fromElement(this, getParent(CtModelImpl.CtRootPackage.class));
+		} catch (CtPathException e) {
+			throw new SpoonException();
```

Comment:
`throw new SpoonException(e);` ... the orgin exception should be used as cause of the SpoonException

Reply:
Yes, no exception should happen, but if it happens because of some mistake in spoon code, then your current implementation will avoid efficient detection of the cause, because the cause of the problem is lost. So just type in `e` into brackets and I am OK with that ;-)

Reply:
> just type in e into brackets

What do you mean? Shall I add `e.printStackTrace();` or change `throw new SpoonException();` into `throw new SpoonException(e);` ?

Reply:
yes please

Reply:
Done in 8cfc1adeff6331dc05d73a316ab6d78d2cc54e7d